Tests/reviews.api.test.js

```diff
@@ -0,0 +1,133 @@
+      .get("/Reviews")
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+
+      });
+  });
+
+  test("/Reviews/getReviews/:productId : it should respond with an array of reviews, respond 200 for valid response ", async () => {
+    var productId = 1000;
+    var mockData = await SeedData.generateSeedData(1, 1);//load 1 review for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviews/${productId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toEqual(true);
+        var insertedData = response.body;
+        expect(insertedData[0].productId).toEqual(productId);
+      });
+
+  });
+
+  test("/Reviews/getReviews/:productId : it should respond with an empty array of reviews, respond 400 for invalid product ID ", async () => {
+    var invalidProductId = 50;
+    var mockData = await SeedData.generateSeedData(1, 1);//load 1 review for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviews/${invalidProductId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(404);
+        expect(Array.isArray(response.body)).toEqual(true);
+        var insertedData = response.body;
+        expect(insertedData[0]).toBeUndefined();
+      });
+
+  });
+
+
+  test("/Reviews/getReviewSummary/:productId : it should respond with an object, respond 200 for valid response ", async () => {
+    var productId = 1000;
+    var mockData = await SeedData.generateSeedData(10, 1);//load 10 customer,  for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviewSummary/${productId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toEqual(false);
+        expect(typeof (response.body)).toEqual('object');
+        var insertedData = response.body;
+        expect.objectContaining({
+          averageRating: expect.any(String),
+          totalRatings: expect.any(Number),
+          fiveStar: expect.any(String),
+          fourStar: expect.any(String),
+          threeStar: expect.any(String),
+          twoStar: expect.any(String),
+          oneStar: expect.any(String)
+        });
+      });
+
+  });
+
+  test("/Reviews/getReviewExcerpts/:productId : it should respond with an array, respond 200 for valid response ", async () => {
+    var productId = 1000;
+    var mockData = await SeedData.generateSeedData(50, 1);//load 10 customer,  for 1 product
+    await Service.insertSeedData(mockData);
+
+    return request(app)
+      .get(`/Reviews/getReviewExcerpts/${productId}`)
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toEqual(true);
+        var insertedData = response.body;
+        expect(insertedData.length).toBeLessThanOrEqual(12);
```

Comment:
Why do you expect the inserted data length to be less than or equal to 12? Maybe put a comment in explaining for your using the number 12.

Reply:
For amazon phrases, they have 12 phrases that shows up in reviews section for search. It can be less than 12 in scenarios where reviews are less than 12 as i am picking up phrases from different reviews. I will add it in comments in next PR.